src/main/kotlin/no/nav/syfo/service/SenderFacade.kt

```diff
@@ -73,38 +75,39 @@ class SenderFacade(
         varsel: ArbeidsgiverNotifikasjonInput,
     ) {
         arbeidsgiverNotifikasjonService.sendNotifikasjon(varsel)
-        lagreUtsendtNarmesteLederVarsel(ARBEIDSGIVERNOTIFIKASJON, varselHendelse, varsel.uuid.toString())
+        lagreUtsendtNarmesteLederVarsel(ARBEIDSGIVERNOTIFIKASJON, varselHendelse, varsel.uuid.toString(), varsel.merkelapp)
     }
 
     fun ferdigstillArbeidsgiverNotifikasjoner(
-        varselHendelse: NarmesteLederHendelse,
-        merkelapp: String
+        varselHendelse: NarmesteLederHendelse
     ) {
-        val eksterneReferanser = database.fetchUtsendtVarsel(
+        val uferdigstilteVarsler = database.fetchUtsendtVarsel(
             varselHendelse.arbeidstakerFnr,
             varselHendelse.orgnummer,
             varselHendelse.type,
             ARBEIDSGIVERNOTIFIKASJON
-        ).eksterneRefUferdigstilteVarsler(varselHendelse.type)
-        for (eksternReferanse in eksterneReferanser) {
-            arbeidsgiverNotifikasjonService.deleteNotifikasjon(
-                merkelapp,
-                eksternReferanse!!
-            )
-            database.setUtsendtVarselToFerdigstilt(eksternReferanse)
+        ).uferdigstilteVarsler(varselHendelse.type)
+        for (varsel in uferdigstilteVarsler) {
+            varsel.eksternReferanse?.let {
+                arbeidsgiverNotifikasjonService.deleteNotifikasjon(
+                    varsel.arbeidsgivernotifikasjonMerkelapp!!,
```

Comment:
Hm, er dette riktig? eksternReferanse?.let, og så kjøre non-null assertion på arbeidsgivernotifikasjonMerkelapp? 

Tenker vi bør prøve å droppe bruk av !! så mye som mulig og heller bruke riktig typing eller "?" eller noe annet

Reply:
Ja, du har rett, jeg var litt usikker på hvordan det skulle håndteres. I praksis er det nok ikke noe problem fordi alle varsler med kanal arbeidsgivernotifikasjon skal ha en merkelapp, men enig at det ikke er så bra. 
Men hvordan er det best å håndtere det da, mon tro? Kaste exception hvis den er null? Silently ignore?

Reply:
I database migrasjonen er merkelappen "nullable":

```ALTER TABLE UTSENDT_VARSEL
    ADD COLUMN arbeidsgivernotifikasjon_merkelapp VARCHAR(50);

UPDATE UTSENDT_VARSEL
set arbeidsgivernotifikasjon_merkelapp = 'Oppfølging'
WHERE kanal = 'ARBEIDSGIVERNOTIFIKASJON'
  and type not in ('NL_DIALOGMOTE_INNKALT', 'NL_DIALOGMOTE_NYTT_TID_STED', 'NL_DIALOGMOTE_REFERAT', 'NL_DIALOGMOTE_AVLYST')
  and ferdigstilt_tidspunkt is null;
UPDATE UTSENDT_VARSEL
set arbeidsgivernotifikasjon_merkelapp = 'Dialogmøte'
WHERE kanal = 'ARBEIDSGIVERNOTIFIKASJON'
  and type in ('NL_DIALOGMOTE_INNKALT', 'NL_DIALOGMOTE_NYTT_TID_STED', 'NL_DIALOGMOTE_REFERAT', 'NL_DIALOGMOTE_AVLYST')
  and ferdigstilt_tidspunkt is null;```
  
 Hadde det vært mulig å sette den til `NOT NULL` med en default verdi 'Unset' eller noe slikt, og deretter kjøre updaten
på tabellen? Da kan man kanskje fjerne ? fra merkelapp typen?

Reply:
... siden update-scriptet har to gjensidig utelukkende predikater som dekker tilfeller der ag-notifikasjon skal ha merkelapp, så vil vel ikke tilfelle 'Unset' dukke opp i praksis der kolonnen brukes.

Reply:
Tenker det også er det beste @torchr89 , slik at typen ikke er nullable

Reply:
Det er bare varsler med kanal = 'ARBEIDSGIVERNOTIFIKASJON' som vil ha denne informasjonen, derfor jeg synes det er litt snålt at den skal være `NOT NULL` i databasen

Reply:
Åja, det er sant

Reply:
Jeg justerte til dette nå, ser det greit ut?
```kotlin
if (varsel.eksternReferanse != null && varsel.arbeidsgivernotifikasjonMerkelapp != null) {
  arbeidsgiverNotifikasjonService.deleteNotifikasjon(
      varsel.arbeidsgivernotifikasjonMerkelapp,
      varsel.eksternReferanse
  )
  database.setUtsendtVarselToFerdigstilt(varsel.eksternReferanse)
} else {
  log.warn("Kan ikke ferdigstille varsel med uuid ${varsel.uuid} fordi den mangler eksternReferanse eller arbeidsgivernotifikasjonMerkelapp")
}
```

Reply:
Mye bedre :) Kunne vurdert log.error, for dette skal vel i teorien ikke kunne skje?